src/DotVVM.Framework.Tests.Common/ResourceManagement/ResourceRepositoryTests.cs

```diff
@@ -25,5 +25,199 @@ public void ResourceRepository_CyclicDependency_Throws()
+        {
+            Assert.ThrowsException<DotvvmResourceException>(() => {
+                var configuration = DotvvmConfiguration.CreateDefault();
+
+                configuration.Resources.Register("6", new NullResource { Dependencies = new[] { "5" } });
+                configuration.Resources.Register("5", new NullResource { Dependencies = new[] { "4" } });
+                configuration.Resources.Register("4", new NullResource { Dependencies = new[] { "3" } });
+                configuration.Resources.Register("3", new NullResource { Dependencies = new[] { "2" } });
+                configuration.Resources.Register("2", new NullResource { Dependencies = new[] { "1" } });
+                configuration.Resources.Register("1", new NullResource { Dependencies = new[] { "6" } });//cyclic
+
+                var manager = configuration.ServiceProvider.GetRequiredService<ResourceManager>();
+                manager.AddRequiredResource("6");
+            });
+        }
+
+        [TestMethod]
+        public void ResourceRepository_CyclicDependency_Structure_5_Throws()
+        {
+            Assert.ThrowsException<DotvvmResourceException>(() => {
+                var configuration = DotvvmConfiguration.CreateDefault();
+                
+                configuration.Resources.Register("3", new NullResource { Dependencies = new[] { "2" } });
+                configuration.Resources.Register("2", new NullResource { Dependencies = new[] { "1" } });
+                configuration.Resources.Register("1", new NullResource { Dependencies = new[] { "3" } });//cyclic
+
+                var manager = configuration.ServiceProvider.GetRequiredService<ResourceManager>();
+                manager.AddRequiredResource("3");
+            });
+        }
+
+        [TestMethod]
+        public void ResourceRepository_CyclicDependency_Structure_6()
+        {
+            Assert.ThrowsException<DotvvmResourceException>(() => {
+                var configuration = DotvvmConfiguration.CreateDefault();
+
+                configuration.Resources.Register("7", new NullResource { Dependencies = new[] { "5" } });
+
+                configuration.Resources.Register("6", new NullResource { Dependencies = new[] { "4", "3" } });
+                configuration.Resources.Register("5", new NullResource { Dependencies = new[] { "3", "1" } });
+
+                configuration.Resources.Register("4", new NullResource { Dependencies = new[] { "1" } });
+                configuration.Resources.Register("3", new NullResource { Dependencies = new[] { "2" } });
+
+                configuration.Resources.Register("2", new NullResource { Dependencies = new[] { "dotvvm" } });
+                configuration.Resources.Register("1", new NullResource { Dependencies = new[] { "dotvvm" } });
+
+                var manager = configuration.ServiceProvider.GetRequiredService<ResourceManager>();
+                manager.AddRequiredResource("3");
```

Comment:
This is not cyclic, why does it throw?

Reply:
@OndrejSulc:
> 6)This method ResourceRepository_CyclicDependency_Structure_6() is not
specified as Throw().

The test framework ignores the name, what makes this test assert that the registration fails is the invocation of `Assert.ThrowsException` method.